File: src/main/java/nieboczek/lifestolen/serializer/base/Serializer.java
package nieboczek.lifestolen.serializer.base;

import nieboczek.lifestolen.serializer.lang.SerializedStringBuilder;
import nieboczek.lifestolen.serializer.lang.TokenStream;

public abstract class Serializer<T> {
    public abstract void serialize(T value, SerializedStringBuilder builder);

    public abstract T deserialize(TokenStream stream);
}

File: src/main/kotlin/nieboczek/lifestolen/serializer/base/Serializer.kt
package nieboczek.lifestolen.serializer.base

import nieboczek.lifestolen.serializer.lang.SerializedStringBuilder
import nieboczek.lifestolen.serializer.lang.TokenStream

abstract class Serializer<T> {
    abstract fun serialize(value: T, builder: SerializedStringBuilder)
    abstract fun deserialize(stream: TokenStream): T
}
